nntrainer/layers/conv2d_layer.cpp

```diff
@@ -396,27 +398,39 @@ void Conv2DLayer::forwarding(RunLayerContext &context, bool training) {
 
   filter_kernel.reshape(filter_dim_squeezed);
 
-  Tensor &im2col_result = context.getTensor(wt_idx[ConvParams::inter_result]);
   /**
    * @todo im2col_result lifespan can be epoch and then setZero can be done
    * just once at the start of training then every iteration
    *
    * @todo even better, allocate in_sub with pad, and set stride for in_sub
    * appropriately
    */
+  Tensor &im2col_result = context.getTensor(wt_idx[ConvParams::inter_result]);
+  im2col_result.setZero();
+
   /**
    * Below sets the pad area values to zero
    * it is faster to do this way than seting selective area to zero
    */
-  im2col_result.setZero();
-  for (unsigned int b = 0; b < in_dim.batch(); ++b) {
-    Tensor out = hidden_.getBatchSlice(b, 1);
-    out.reshape({filter_size, out_dim.width() * out_dim.height()});
+  auto forwarding_job = [&](unsigned int s, unsigned int e, void *user_data) {
+    Tensor result = Tensor(calcCol2ImOutputDim(out_dim, filter_dim));
+    result.setZero();
```

Comment:
I have question.
Is there a possibility that calculated output dimention (calcCol2ImOutputDim) is differ from im2col result?
I'm having a question about setting zero for result.

Reply:
The reason of setting zero is padding.  padded part of result needs to be set zero.